### DISCUSSION TITLE
[FEATURE REQUEST] add an option for generating all the variations from 1 seed from all the classifier or steps or both

### DISCUSSION THREAD

**Discussion by search620** [hlky/stable-diffusion-webui#202] [Feature suggestions]
add an option for generating all the variations from 1 seed from all the classifier or steps or both 

It would be really great to have the ability to select a seed and then manually choose how much variation to generate from it by classifier or steps or both in a row. 

At the current status if I want to do it then I have to do it by hand step by step manually (time-consuming)
The reason I'm interested in a feature like this is that sometimes there are good seeds and I noticed that If I played a little bit with the sliders of the steps or the classifier then the results can be a great outcome.

What do you think about this type of feature?

**Comment by hlky** [comment]
Yes I thought about this, I would probably make it a standalone script to avoid complicating the main ui and keep it out the way of people who don't need it

I have noticed the same as you different cfg and steps on the same seed can be different results.

My thoughts were if you found an image, had the seed and prompt, but not the cfg, you could 'bruteforce' by trying all the possible cfg values, this might be useful if the image file is lost, or an image you find you like that has since been watermarked, or something like that. Same thing basically but different usage

Btw I am thinking about moving feature requests to discussions, each feature request can have a discussion and then there can be a main discussion for new features with a list.
In fact, I will move this to discussions now.

**Comment by search620** [comment]
I hope people who are interested in this feature will comment here. 

You can open this feature in a separate tab. It is also possible to arrange the UI in a slightly more convenient way Like for example tabs that open and close when you want to see or hide all the settings (More or less as of now, just arranged a little differently).  
![2022-08-28_22-29-20](https://user-images.githubusercontent.com/56544202/187091526-3992765d-a048-4f06-8c89-55aef1aa4dd6.png)


If you want recommendations, let me know and I'll be happy to give them.
The ideas for improvement do not end with me, believe me. But I concentrate on what is important.

Thank you for all really appreciated the effort and time you invest in it.

**Comment by hlky** [comment]
The UI is subject to change for now, we'll hopefully decide on a final choice of layout soon.

I'm leaning toward making most advanced features their own tab e.g. gfpgan, upscalers, the advanced image editor, and things like this

**Reply by search620** [reply]
So to be organized is always for the better. Tabs for each multi-function thing Is Good. I must say that the basic thing like upscale and face correction need to be as they are now just with a little change... each one of the models can save separately as for now (this is good). all that needed to do is to put a check box that combines all the upscale and face corrections into 1 merged image 

some users want all these files and some users just want the final results.

I hope I was clear if something doesn't smooth for you please note me.

Thanks!

**Reply by hlky** [reply]
Having both gfpgan and esrgan enabled should produce only 1 file, it was fixed earlier, but I just checked and one line must have been reverted by a pr, should be fixed now, and when only 1 image is being produced the grid won't be added to the gradio output gallery 

#189  related

**Reply by hlky** [reply]
Well, technically two files unless you also disable the 'save individual samples' option

**Reply by JoshuaKimsey** [reply]
I definitely support the idea of advanced features being on one tab, probably called post-processing(?), since we really don't know how many more GAN's are being made. Maybe one for hair, maybe one for foliage, heck maybe even one for hands! (I bet a lot of people would like that)

But it's definitely good future proofing to go ahead and do this.
